packages/server-core/src/bot/bot/bot.hooks.ts

```diff
@@ -39,7 +44,29 @@ import {
   botResolver
 } from '../../bot/bot/bot.resolvers'
 import authenticate from '../../hooks/authenticate'
+import persistData from '../../hooks/persist-data'
 import verifyScope from '../../hooks/verify-scope'
+import { BotService } from './bot.class'
+
+async function sanitizeInstanceId(context: HookContext<BotService>) {
+  if (Array.isArray(context.data) || context.method !== 'create') {
+    throw new BadRequest(`${context.path} service only works for single object create`)
+  }
+
+  context.data!.instanceId = context.data?.instanceId ?? ('' as InstanceID)
+}
+
+async function createAndPopulateBotCommands(context: HookContext<BotService>) {
+  const botCommands: BotCommandType[] = []
+  for (const commandData of context.actualData.botCommands) {
+    const command = await context.app.service(botCommandPath).create({
+      ...commandData,
+      botId: (context.result as BotType).id
+    })
+    botCommands.push(command)
+  }
+  ;(context.result as BotType).botCommands = botCommands
```

Comment:
This hook can be made dynamic for array and single object by introducing a process() method and then calling it like done in `createSelfOwner` hook of `packages/server-core/src/social/channel/channel.hooks.ts`